### PR TITLE
metadata: add double-click + reset button to reset prefs

### PR DESCRIPTION
**Pull request by phweyland**
- double-click on label to reset the metadata
- button reset to set default prefs (metadata visible and private)

Could be revised following conclusions of #7859.

### CODE REVIEW

File: src/common/metadata.c [hunk 1]
```diff
@@ -184,6 +184,24 @@ void dt_metadata_init()
   }
 }
 
+void dt_metadata_reset()
+{
+  for(unsigned int i = 0; i < DT_METADATA_NUMBER; i++)
+  {
+    const int type = dt_metadata_get_type(i);
+    const char *name = (gchar *)dt_metadata_get_name(i);
+    char *setting = dt_util_dstrcat(NULL, "plugins/lighttable/metadata/%s_flag", name);
```
Comment (TurboGit) [inline src/common/metadata.c:193]: need to be freed

**Review by TurboGit — CHANGES_REQUESTED** [review]
A memory leak to fix.

File: src/common/metadata.c (hunk 1 above)
Comment (phweyland) [reply src/common/metadata.c:193]: Yes! I fix that.

**Comment by phweyland** [comment]
Relative to discussion in #7859 ...

If I follow our both statements I should make the following changes to this module:
- remove clear button and replace it by reset button (=> metadata values)
- add a reset button in the module pref dialog (=> metadata visibility, ...).

How does that sound ?

**Comment by TurboGit** [comment]
> How does that sound ?

Correct on both counts to me.

**Comment by phweyland** [comment]
I've disabled the reset button, added a "default" button in prefs dialog.
For the user, the advantage is to be able to cancel the reset action if he doesn't feels secure.

If this is ok like this, I'll do the same for metadata_view for consistency.

**Comment by TurboGit** [comment]
> I've disabled the reset button, added a "default" button in prefs dialog.

Hum... I still prefer having the reset button to clear the entries. We have a clear button [apply] that clearly shows that unit it is clicked the metadata won't be affecting the selected images.

**Comment by phweyland** [comment]
> I still prefer having the reset button to clear the entries.

I had a preference for my last proposal, but this is good too. Back to first solution without memory leak.

**Review by TurboGit — COMMENTED** [review]
The reset button does nothing on my side with this last version.

**Comment by phweyland** [comment]
Are you sure ? It works for me.
It set back visibility and private flags to default values.
But doesn't touch to metadata values of the selected images.

**Comment by TurboGit** [comment]
> But doesn't touch to metadata values of the selected images.

Ok, so we have misunderstanding. I prefer a version I tested some time ago plus the proposal you had partly implemeneted IIRC:
- reset button : clear all metadata fields
- the [clear] button can then be removed
- on the metadata settings dialog a new button is added to reset to default values (I'm sure I had seen / tested that).

This seems more consistent.

**Comment by TurboGit** [comment]
The point is that really reset button should clear what is visible or at least at the same level but not into sub-dialog.

**Comment by phweyland** [comment]
Should be better now :wink:

**Comment by TurboGit** [comment]
> However I've commented the acceleration for clear button. Is there an equivalent for reset button ?

Yes, all reset button can be assigned a key in the accels editor.

**Review by TurboGit — APPROVED** [review]
Thanks !

**Pull request merged**